File: api/models/Group.js
const mongoose = require("mongoose")

const groupSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
    subject: {
        type: String,
        required: true,
    },
    ownerId: {
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User',
        required: true,
    },
    membersID: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
    }],
    inviteCode: {
        type: String,
        required: true,
    },
    createDate: {
        type: Date,
        default: Date.now
    }
})
module.exports = mongoose.model('Group', groupSchema);

File: app/lib/models/group.dart
class Group {
  String? id;
  String? name;
  String? description;
  String? subject;
  dynamic ownerId;
  List<String>? membersID;
  String? inviteCode;
  String? createDate;
  String? imgGroup;

  Group({
    this.id,
    this.name,
    this.description,
    this.subject,
    this.ownerId,
    this.membersID,
    this.createDate,
    this.inviteCode,
    this.imgGroup,
  });

  Group.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    name = json['name'] is Map
        ? json['name']['vi'] ?? json['name'].values.first
        : json['name'];
    description = json['description'];
    subject = json['subject'];
    ownerId = json['ownerId'];
    membersID = (json['membersID'] as List<dynamic>?)
        ?.map((member) {
          if (member is String) return member;
          if (member is Map && member['_id'] != null)
            return member['_id'].toString();
          return '';
        })
        .where((id) => id.isNotEmpty)
        .toList();
    createDate = json['createDate'];
    inviteCode = json['inviteCode'];
    imgGroup = json['imgGroup'];
  }


  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'subject': subject,
      'ownerId': ownerId,
      'membersID': membersID,
      'createDate': createDate,
      'inviteCode': inviteCode,
      'imgGroup': imgGroup,
    };
  }
}
